automation/taskcluster/lib/tasks.py

```diff
@@ -436,6 +452,115 @@ def craft_push_task(
             },
         )
 
+    def craft_raptor_tp6m_cold_task(self, for_suite):
+
+        def craft_function(signing_task_id, mozharness_task_id, variant, gecko_revision, force_run_on_64_bit_device=False):
+            return self._craft_raptor_task(
+                signing_task_id,
+                mozharness_task_id,
+                variant,
+                gecko_revision,
+                name_prefix='raptor tp6m-cold-{}'.format(for_suite),
+                description='Raptor tp6m cold on Fenix',
+                test_name='raptor-tp6m-cold-{}'.format(for_suite),
+                job_symbol='tp6m-c-{}'.format(for_suite),
+                force_run_on_64_bit_device=force_run_on_64_bit_device,
+            )
+        return craft_function
+
+    def _craft_raptor_task(
+        self,
+        signing_task_id,
+        mozharness_task_id,
+        variant,
+        gecko_revision,
+        name_prefix,
+        description,
+        test_name,
+        job_symbol,
+        group_symbol=None,
+        extra_test_args=None,
+        force_run_on_64_bit_device=False,
+    ):
+        extra_test_args = [] if extra_test_args is None else extra_test_args
+        apk_location = '{}/{}/artifacts/{}'.format(
+            _DEFAULT_TASK_URL, signing_task_id, DEFAULT_APK_ARTIFACT_LOCATION
+        )
+        architecture, _ = get_architecture_and_build_type_from_variant(variant)
+        worker_type = 'gecko-t-ap-perf-p2' if force_run_on_64_bit_device or architecture == 'aarch64' else 'gecko-t-ap-perf-g5'
```

Comment:
we need to fix this (as a followup is ok) for both refbrow and fenix- we only use gecko-t-ap-perf-g5, but for power/battery tests we need gecko-t-ap-batt-g5|p2.  It isn't a priority to have speedometer power running, but having the plumbing so we can edit the test name/definition and it just works would be really nice.

Reply:
Good catch! I'll handle it in a followup.